core/validatorapi/grpc/validator.go

```diff
@@ -0,0 +1,512 @@
+		var blinded *eth2api.VersionedSignedBlindedBeaconBlock
+		// blinded := convert(block)
+		blockRoot, _ = blinded.Root()
+
+		_ = s.eth2Cl.SubmitBlindedBeaconBlock(ctx, blinded)
+	} else {
+		var bblock *spec.VersionedSignedBeaconBlock
+		// bblock := convert(block)
+		blockRoot, _ = bblock.Root()
+
+		_ = s.eth2Cl.SubmitBeaconBlock(ctx, bblock)
+	}
+
+	return &prysmpb.ProposeResponse{
+		BlockRoot: blockRoot[:],
+	}, nil
+}
+
+// PrepareBeaconProposer 🎉 can proxy to valiatgorAPI
+func (s *BeaconNodeValidatorServer) PrepareBeaconProposer(ctx context.Context, req *prysmpb.PrepareBeaconProposerRequest) (*empty.Empty, error) {
+	var preps []*eth2v1.ProposalPreparation
+	for _, recipient := range req.Recipients {
+		var addr bellatrix.ExecutionAddress
+		copy(addr[:], recipient.FeeRecipient)
+
+		preps = append(preps, &eth2v1.ProposalPreparation{
+			ValidatorIndex: eth2p0.ValidatorIndex(recipient.ValidatorIndex),
+			FeeRecipient:   addr,
+		})
+	}
+
+	return &empty.Empty{}, s.prepareSubmitter.SubmitProposalPreparations(ctx, preps)
+}
+
+// GetAttestationData 🎉 can proxy to valiatgorAPI
+func (s *BeaconNodeValidatorServer) GetAttestationData(ctx context.Context, req *prysmpb.AttestationDataRequest) (*prysmpb.AttestationData, error) {
+	data, _ := s.eth2Cl.AttestationData(ctx, eth2p0.Slot(req.Slot), eth2p0.CommitteeIndex(req.CommitteeIndex))
+
+	return &prysmpb.AttestationData{
+		Slot:            pb.Slot(data.Slot),
+		CommitteeIndex:  pb.CommitteeIndex(data.Index),
+		BeaconBlockRoot: data.BeaconBlockRoot[:],
+		Source:          convert(data.Source).(*prysmpb.Checkpoint),
+		Target:          convert(data.Target).(*prysmpb.Checkpoint),
+	}, nil
+}
+
+// ProposeAttestation 🎉 can proxy to valiatgorAPI
+func (s *BeaconNodeValidatorServer) ProposeAttestation(ctx context.Context, attestation *prysmpb.Attestation) (*prysmpb.AttestResponse, error) {
+	att := convert(attestation).(*eth2p0.Attestation)
+	root, _ := att.Data.HashTreeRoot()
+
+	_ = s.eth2Cl.SubmitAttestations(ctx, []*eth2p0.Attestation{att})
+
+	return &prysmpb.AttestResponse{AttestationDataRoot: root[:]}, nil
+}
+
+// SubmitAggregateSelectionProof can use integrated but probably 🆙 requires upgrade along with SubscribeCommitteeSubnets.
+func (s *BeaconNodeValidatorServer) SubmitAggregateSelectionProof(ctx context.Context, req *prysmpb.AggregateSelectionRequest) (*prysmpb.AggregateSelectionResponse, error) {
+	// TODO(corver): This is an interesting alternative to our v2 API change.
+	//  The VC provides the partial selection proof and the BN returns the complete aggregation and proof
+	//  for the VC to sign. The VC basically doesn't need to be any logic.
+
+	// Assuming we already completed `core.DutyPrepareAggregator` for this slot.
+	aggSig, _ := s.aggSigDB.Await(ctx, core.NewPrepareAggregatorDuty(int64(req.Slot)), toGroupPubkey(req.PublicKey))
+
+	// TODO(corver): We will need to add a different API to dutyDB to query the DutyAggregator data by pubkey instead of attestation hash.
```

Comment:
another alternative can be to query attestation and then use its data root to query aggregate attestation, but this will require 2 queries to DutyDB.

Reply:
sure